src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,194 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int NEW_SIZE_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * table.length);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        int backet = getHash(key);
+        if (table[backet] == null) {
+            table[backet] = newNode;
+            size++;
+            return;
+        } else {
+            putWithCollision(newNode);
+        }
+        if (size == threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        V value;
+        Node<K, V> currentNode;
+        for (Node<K, V> nodeForGetValue : table) {
+            currentNode = nodeForGetValue;
+            while (currentNode != null) {
+                if (getKey(currentNode) == key
+                        || (getKey(currentNode)) != null
+                        && getKey(currentNode).equals(key)) {
+                    value = getValueFromNode(currentNode);
+                    return value;
+                }
+                currentNode = currentNode.next;
```

Comment:
you should find out bucket by key and based on this start looking for needed one